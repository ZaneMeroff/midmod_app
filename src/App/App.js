import React, { Component } from 'react';
import Form from '../Form/Form';
import CardArea from '../CardArea/CardArea';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: [ {} ]
    }
  }

  postResOnDOM = reservation => {
    this.setState({reservations: [...this.state.reservations, reservation]})
  }

  componentDidMount = () => {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(response => response.json())
      .then(data => this.setState({reservations: data}))
  }

  postToAPI = (name, date, time, number ) => {
    const options = {
      method: 'POST',
      body: JSON.stringify({
        id: this.state.reservations.length + 1,
        name: name,
        date: date,
        time: time,
        number: number
      }),
      headers: {'Content-Type': 'application/json'}
    }
    fetch('http://localhost:3001/api/v1/reservations', options)
      .then(response => response.json())
      .then(newRes => this.setState({reservations: [...this.state.reservations, newRes]}))
    }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form postToAPI={this.postToAPI} res={this.state.reservations}/>
        </div>
        <div className='resy-container'>
          <CardArea cards={this.state.reservations} />
        </div>
      </div>
    )
  }
}

export default App;
