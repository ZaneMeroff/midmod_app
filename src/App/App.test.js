import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let mockRes = [{
    name: 'fakeName',
    date: 'fakeDate',
    time: 'fakeTime',
    number: 'fakeNumber',
    id: 'fakeId',
  }]

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when postResOnDOM is invoked', () => {
    wrapper.instance().postResOnDOM(mockRes)
    expect(wrapper.state('reservations')).toEqual(mockRes)
  });

})
