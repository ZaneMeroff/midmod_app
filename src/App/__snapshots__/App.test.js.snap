// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      postToAPI={[Function]}
      res={
        Array [
          Object {},
        ]
      }
    />
  </div>
  <div
    className="resy-container"
  >
    <CardArea
      cards={
        Array [
          Object {},
        ]
      }
    />
  </div>
</div>
`;
